src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/WhenProcessingEnvelopeTest.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.core.PathUtility;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.StorageConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {StorageConfiguration.class})
+public class WhenProcessingEnvelopeTest {
```

Comment:
Why `When...`?

Reply:
given/when/then is typically represented by various clauses or sometimes even comments. Class names by convention represent the subject of testing, i.e. class/component or a piece of its functionality. This is the convention we're using everywhere.